pkg/ddc/jindo/dataset.go

```diff
@@ -1,31 +1,128 @@
 package jindo
 
 import (
+	"context"
+	"reflect"
+
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/retry"
 )
 
 func (e *JindoEngine) UpdateDatasetStatus(phase datav1alpha1.DatasetPhase) (err error) {
-	/*err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+	// 1. update the runtime status
+	runtime, err := e.getRuntime()
+	if err != nil {
+		return err
+	}
+
+	// 2.update the dataset status
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
 		dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
 		if err != nil {
 			return err
 		}
 		datasetToUpdate := dataset.DeepCopy()
+		var cond datav1alpha1.DatasetCondition
+
+		switch phase {
+		case datav1alpha1.BoundDatasetPhase:
+			cond = utils.NewDatasetCondition(datav1alpha1.DatasetReady, datav1alpha1.DatasetReadyReason,
+				"The ddc runtime is ready.",
+				corev1.ConditionTrue)
+		case datav1alpha1.FailedDatasetPhase:
+			cond = utils.NewDatasetCondition(datav1alpha1.DatasetReady, datav1alpha1.DatasetReadyReason,
+				"The ddc runtime is not ready.",
+				corev1.ConditionFalse)
+		default:
+			cond = utils.NewDatasetCondition(datav1alpha1.DatasetReady, datav1alpha1.DatasetReadyReason,
+				"The ddc runtime is unknown.",
+				corev1.ConditionFalse)
+		}
+
 		datasetToUpdate.Status.Phase = phase
-		err = e.Client.Status().Update(context.TODO(), datasetToUpdate)
-		if err != nil {
-			e.Log.Error(err, "Update dataset")
-			return err
+		datasetToUpdate.Status.Conditions = utils.UpdateDatasetCondition(datasetToUpdate.Status.Conditions,
+			cond)
+
+		datasetToUpdate.Status.CacheStates = runtime.Status.CacheStates
+
+		e.Log.Info("the dataset status", "status", datasetToUpdate.Status)
+
+		if !reflect.DeepEqual(dataset.Status, datasetToUpdate.Status) {
+			err = e.Client.Status().Update(context.TODO(), datasetToUpdate)
+			if err != nil {
+				e.Log.Error(err, "Update dataset")
+				return err
+			}
 		}
+
 		return nil
-	})*/
-	return nil
+	})
+
+	if err != nil {
+		e.Log.Error(err, "Update dataset")
+		return err
+	}
+
+	return
 }
 
 func (e *JindoEngine) UpdateCacheOfDataset() (err error) {
```

Comment:
`je` is better ?  [receiver-names](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names)